packages/backend-api/integration/integrate.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+INTEGRATION_DIR="$( cd "$(dirname "$0")" ; pwd -P )"
+PROJECT_BASE_PATH=`[ -d "$PWD/project-base" ] && echo "$PWD/project-base" || echo "$PWD"`
+
+## TODO check that the ${PROJECT_BASE_PATH} is correct project-base
+
+echo "Copying FOS REST configuration..."
+cp ${INTEGRATION_DIR}/app/config/packages/fos_rest.yml ${PROJECT_BASE_PATH}/app/config/packages/fos_rest.yml || exit 1
+echo "Done"
+
+echo "Copying OAuth2 configuration..."
+cp ${INTEGRATION_DIR}/app/config/packages/trikoder_oauth2.yml ${PROJECT_BASE_PATH}/app/config/packages/trikoder_oauth2.yml || exit 1
+mkdir -p ${PROJECT_BASE_PATH}/app/config/packages/oauth2 || exit 1
+cp ${INTEGRATION_DIR}/app/config/packages/oauth2/.gitignore ${PROJECT_BASE_PATH}/app/config/packages/oauth2/.gitignore || exit 1
+cp ${INTEGRATION_DIR}/app/config/packages/oauth2/parameters_oauth.yml.dist ${PROJECT_BASE_PATH}/app/config/packages/oauth2/parameters_oauth.yml.dist || exit 1
+echo "Done"
+
+echo "Creating directory src/Shopsys/ShopBundle/Controller/Api/V1..."
+mkdir -p ${PROJECT_BASE_PATH}/src/Shopsys/ShopBundle/Controller/Api/V1
+touch ${PROJECT_BASE_PATH}/src/Shopsys/ShopBundle/Controller/Api/V1/.gitkeep
+echo "Done"
+
+echo "Copying tests..."
+cp ${INTEGRATION_DIR}/tests/ShopBundle/Smoke/BackendApiTest.php ${PROJECT_BASE_PATH}/tests/ShopBundle/Smoke/BackendApiTest.php || exit 1
+cp ${INTEGRATION_DIR}/tests/ShopBundle/Test/OauthTestCase.php ${PROJECT_BASE_PATH}/tests/ShopBundle/Test/OauthTestCase.php || exit 1
+echo "Done"
+
+function apply_patch () {
+    if [ -z $1 ]
+    then
+        echo "Please provide path as first parameter of apply_patch function"
+        exit 0
+    fi
+    
+    local FILE_PATH=$1
+    local SOURCE_FILE_PATH=$1
+
+    if [ ! -z $2 ]
+    then
+        SOURCE_FILE_PATH=$2
+    fi
+
+    echo "Applying patch for ${FILE_PATH}..."
+    local PATCH_DRY_RUN=`patch -st --dry-run ${PROJECT_BASE_PATH}/${FILE_PATH} ${INTEGRATION_DIR}/${SOURCE_FILE_PATH}.patch`
+    local PATCH_REVERSED=`grep "Reversed" <<< ${PATCH_DRY_RUN}`
+    local PATCH_FAIL=`grep "FAILED" <<< ${PATCH_DRY_RUN}`
+
+    if [ ! -z "${PATCH_REVERSED}" ]
+    then
+        echo "Applied patch detected: patch for ${FILE_PATH} already applied. Doing nothing"
+    elif [ ! -z "${PATCH_FAIL}" ]
+    then
+        echo ${PATCH_FAIL}
+        echo "Patch for ${FILE_PATH} cannot be applied!"
+        exit 1
+    else
+        patch -st ${PROJECT_BASE_PATH}/${FILE_PATH} ${INTEGRATION_DIR}/${SOURCE_FILE_PATH}.patch
+        echo "Done"
+    fi
+}
+
+apply_patch "app/config/parameters_common.yml"
+apply_patch "app/config/routing.yml"
+apply_patch "app/config/packages/security.yml"
+apply_patch "src/Shopsys/ShopBundle/Resources/config/routing.yml"
+apply_patch "app/AppKernel.php"
+
+# if running the monorepo integration, apply monorepo patches
+if [ "$1" == "monorepo" ]
+then
+    apply_patch "composer.json" "monorepo.composer.json"
```

Comment:
there is project_base_path 
i am not sure whether this will modify composer.json in monorepo which is in parent folder of project-base

Reply:
this will modify files inside the `project-base` directory... Should I change anything? Or explain it somehow?

Reply:
hmm if we apply patch for composer.json so i think we should run `composer-dev` or `composer-prod`

Reply:
not when running from monorepo (there are already all depencencies)... And from project-base? Yes, we should. I have just feeling it shouldn't be part of the integration script and I'd like to add it to install instructions (one argument is that in the install script we have no idea wheather `dev` or `prod` should be run)

Reply:
... added to installation instructions